mitmproxy/controller.py

```diff
@@ -45,38 +91,12 @@ def tick(self, timeout):
         return changed
 
     def shutdown(self):
-        self.should_exit.set()
-
-
-class ServerMaster(Master):
```

Comment:
There's no need for multiple layers of abstraction here - the very simple event loop is never used, and in almost all circumstances the server will be running in a different thread. We have many such unused layers in the codebase at the moment that I'd like to cut out.